qutebrowser/mainwindow/tabwidget.py

```diff
@@ -451,11 +483,33 @@ def tabSizeHint(self, index):
             # get scroll buttons as soon as needed.
             size = minimum_size
         else:
+            tab_width_pinned_conf = config.get('tabs', 'pinned-width')
+
+            try:
+                pinned = self.tab_data(index, 'pinned')
+            except KeyError:
+                pass
+            else:
+                if pinned:
+                    size = QSize(tab_width_pinned_conf, height)
+                    qtutils.ensure_valid(size)
+                    return size
+
             # If we *do* have enough space, tabs should occupy the whole window
-            # width.
-            width = self.width() / self.count()
-            # If width is not divisible by count, add a pixel to some tabs so
-            # that there is no ugly leftover space.
+            # width. Also taken in consideration the reduced space necessary
+            #for the pinned tabs.
+            #WORKAROUND: During shutdown the self.count goes down,
```

Comment:
The `WORKAROUND` tag is mainly for workarounds for Qt/PyQt bugs - I think we should either track this down, or just leave the comment without that.